cpp/docs/common/Future__RETURN__NATIVE_PROMISE__HELPER__.adoc

```diff
@@ -0,0 +1,50 @@
+[#_Future__RETURN__NATIVE_PROMISE__HELPER_]
+=== Future< RETURN, NATIVE_PROMISE, HELPER >
```

Comment:
None of the shortened typedef's are documented. Should they be?
They do appear in the return types of methods. 

I can also refactor + specialise the classes a bit to reduce it down to the C++ type only as template argument.